tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -1245,6 +1245,17 @@ Status StorageManager::get_fragment_info(
     if (!sparse)
       array_schema->domain()->expand_to_tiles(&expanded_non_empty_domain);
 
+    // would prefer to not create another FragmentMeta object
+    // but fragment_metadata/meta only exists within the scope of this function?
+    // use tdb_shared_ptr?
+    FragmentMetadata* cmeta = tdb_new(
```

Comment:
Constructing a `FragmentMetadata` instance does not necessarily populate it with the on-disk data. It needs to be loaded with `FragmentMetadata::load`, otherwise `load_rtree` will not load anything. I believe this is the issue you are encountering.

The right thing to do here is to re-use the `meta` on L1238 since it already loaded. While looking at this code, it looks like `meta` already leaks so it's difficult for me to recommend what to do here right now.

Reply:
OK, I see. Thanks for explaining. 

When I pass `meta` to the `SingleFragmentInfo`, the code errors out and does not seem point to a valid `FragmentMetadata` object anymore? When I call `meta->uri()` or` meta->load_rtree()` in the `FragmentInfo` class, those calls error out right away. But when I pass in `cmeta`, at the very least, those calls don't error out. That does confuse me since `array_open_for_reads` calls ` tdb_new(FragmentMetadata, ...)`, but I didn't find an associated `tdb_delete` anywhere. So `meta` here should exist outside the scope of this function, but I don't know why I can't get it to work.

Reply:
Here is what we need to do:

1. `FragmentInfo::load()` must actually open the array without loading fragments (we have such a function in `StorageManager`) and keep an open array object in `FragmentInfo`. 
2. In [this line](https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/fragment/fragment_info.cc#L442) we need to invoke `get_fragment_info` passing the open array object instead of the URI. This function needs refactoring and should never be getting the array URI as input. This will fix some issues we have with consolidation as well.